platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/events/SessionReplayTarget.kt

```diff
@@ -27,37 +26,32 @@ import io.bitdrift.capture.replay.internal.EncodedScreenMetrics
 import io.bitdrift.capture.replay.internal.FilteredCapture
 
 // Controls the replay feature
-internal class ReplayScreenLogger(
+internal class SessionReplayTarget(
+    configuration: SessionReplayConfiguration,
     errorHandler: ErrorHandler,
-    private val context: Context,
+    context: Context,
     private val logger: LoggerImpl,
-    private val processLifecycleOwner: LifecycleOwner,
-    private val mainThreadHandler: MainThreadHandler = MainThreadHandler(),
-    runtime: Runtime,
-    configuration: SessionReplayConfiguration,
-) : LifecycleEventObserver, ReplayLogger {
-
-    private val replayModule: ReplayModule = ReplayModule(errorHandler, this, configuration, runtime)
-
-    fun start() {
-        mainThreadHandler.run {
-            processLifecycleOwner.lifecycle.addObserver(this)
-        }
-    }
+    mainThreadHandler: MainThreadHandler = MainThreadHandler(),
+) : ISessionReplayTarget, ReplayLogger {
+    internal var runtime: Runtime? = null
+    private val replayModule: ReplayModule = ReplayModule(
+        errorHandler,
+        this,
+        configuration,
+        context,
+        mainThreadHandler,
+    )
 
-    fun stop() {
-        mainThreadHandler.run {
-            processLifecycleOwner.lifecycle.removeObserver(this)
-        }
+    override fun captureScreen() {
+        val skipReplayComposeViews = runtime?.isEnabled(RuntimeFeature.SESSION_REPLAY_COMPOSE)
```

Comment:
why is runtime nullable? shouldn't it just be passed at construction time

Reply:
That 2nd TODO seems more relevant now that none of the recurrent loggers are managed at the platform level (resource utilization and now replay). Hopefully we can get to that todo next.